package Nursultan;

import java.util.List;
import java.util.function.Consumer;
import minecraft.class01237;
import minecraft.class01421;
import minecraft.class01434;
import minecraft.class01590;
import minecraft.class03063;
import minecraft.class04790;
import minecraft.class06202;
import minecraft.class06889;
import minecraft.class06959;
import minecraft.class07049;
import minecraft.class08133;

public class class11890 {
   private static byte[] y;
   public static Object[] N;
   private static String[] i;

   private static void L() {
      N = new Object[y[0]];
   }

   private class11890() {
      throw new UnsupportedOperationException(i[0]);
   }

   static {
      N();
      y();
      u();
      L();
   }

   private static void u() {
      i = new String[1];
      i[0] = "This is a utility class and cannot be instantiated";
   }

   private static void y() {
      y = new byte[1];
      y[0] = 4;
   }

   private static void N() {
   }

   public static List<class11904> N(Consumer<class01237> var0) {
      class06202 var1 = class06202.Nq();
      if ((class08133)N[3] == null) {
         class04790 var8 = new class04790();
         N[0] = var8;
         class11913 var9 = new class11913();
         N[1] = var9;
         class11913 var10 = new class11913();
         N[2] = var10;
         class08133 var11 = new class08133((class04790)N[0], var1.yU(), (class11913)N[1], var1.yW(), new class01434(), (class11913)N[2], (class01590)var1.i_3);
         N[3] = var11;
      }

      List<class11904> var2;
      try {
         var0.accept((class04790)N[0]);
         ((class08133)N[3]).N();
         var2 = ((class11913)N[1]).y();
      } finally {
         ((class11913)N[1]).N();
         ((class11913)N[2]).N();
         ((class04790)N[0]).N();
      }

      return var2;
   }

   public static List<class11904> N(class07049 var0, class06889 var1, float var2) {
      class06202 var3 = class06202.Nq();
      class06959 var4 = ((class03063)var3.B_2).B.N;
      return N(var5 -> ((class11792)var3.Ng()).N(var0, var4, var1.M, var1.B, var1.Z, var2, new class01421(), var5));
   }
}
